File: scripts/generate_edge_voices.py
import asyncio
import edge_tts
import os

OUTPUT_DIR = os.path.join(os.path.dirname(__file__), '..', 'public', 'voices', 'female')
if not os.path.exists(OUTPUT_DIR):
    os.makedirs(OUTPUT_DIR)

VOICE = "si-LK-ThiliniNeural"
RATE = "+10%"

phrases = {
    "niyamai.mp3": "නියමයි",
    "patta.mp3": "පට්ට",
    "elakiri.mp3": "එළකිරි",
    "wedak_na.mp3": "වැඩක් නෑ කතා කරලා",
    "win.mp3": "දින්නා ජයවේවා",
    "lose.mp3": "අයියෝ පරාදයි"
}

async def generate():
    for filename, text in phrases.items():
        dest = os.path.join(OUTPUT_DIR, filename)
        communicate = edge_tts.Communicate(text, VOICE, rate=RATE)
        await communicate.save(dest)
        print(f"Saved: {filename}")

if __name__ == "__main__":
    asyncio.run(generate())

File: scripts/generate_edge_voices.js
import { EdgeTTS } from 'edge-tts';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const outputDir = path.join(__dirname, '../public/voices');

if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

// "si-LK-SameeraNeural" is a male Sinhala voice from Microsoft Edge TTS
const tts = new EdgeTTS({
  voice: 'si-LK-SameeraNeural',
  rate: '+15%', // Faster for more energy!
  pitch: '+0Hz',
});

const phrases = [
  { text: "නියමයි", file: "niyamai.mp3" },
  { text: "පට්ට", file: "patta.mp3" },
  { text: "එළකිරි", file: "elakiri.mp3" },
  { text: "වැඩක් නෑ කතා කරලා", file: "wedak_na.mp3" },
  { text: "දින්නා ජයවේවා", file: "win.mp3" },
  { text: "අයියෝ පරාදයි", file: "lose.mp3" }
];

async function generateAll() {
  for (const phrase of phrases) {
    try {
      const dest = path.join(outputDir, phrase.file);
      await tts.ttsPromise(phrase.text, dest);
      console.log(`Saved: ${phrase.file}`);
      // Wait a bit
      await new Promise(r => setTimeout(r, 500));
    } catch (err) {
      console.error(`Error for ${phrase.file}:`, err);
    }
  }
}

generateAll();
